build_tools/testing/generate_cmake_e2e_test_artifacts_suite.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+## Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Generates a CMake file to build test artifacts."""
+
+import itertools
+import sys
+import pathlib
+import argparse
+
+# Add build_tools python dir to the search path.
+sys.path.insert(0, str(pathlib.Path(__file__).parent.with_name("python")))
+
+from e2e_test_artifacts.cmake_generator import model_rule_generator, iree_rule_generator
+import e2e_test_artifacts.artifacts
+
+# CMake variable name to store IREE package name.
+PACKAGE_NAME_CMAKE_VARIABLE = "PACKAGE_NAME"
+ROOT_ARTIFACTS_DIR_CMAKE_VARIABLE = "ROOT_ARTIFACTS_DIR"
+
+GENERATED_E2E_TEST_FETCH_MODELS_CMAKE_FILENAMAE = "generated_e2e_test_fetch_models.cmake"
+GENERATED_E2E_TEST_IREE_ARTIFACTS_CMAKE_FILENAME = "generated_e2e_test_iree_artifacts.cmake"
+
+
+def parse_arguments():
+  """Parses command-line options."""
+
+  parser = argparse.ArgumentParser()
+  parser.add_argument("--output_dir",
+                      required=True,
+                      help="Dir path to write the generated cmake files.")
+
+  return parser.parse_args()
+
+
+def main(args: argparse.Namespace):
+  artifacts_root = (
+      e2e_test_artifacts.artifacts.generate_default_artifacts_root())
+
+  root_path = pathlib.PurePath(f"${{{ROOT_ARTIFACTS_DIR_CMAKE_VARIABLE}}}")
```

Comment:
Given the use of curly braces here, maybe it would be better to use something other than f-strings for clarity?

```suggestion
  root_path = pathlib.PurePath("${%s}" % ROOT_ARTIFACTS_DIR_CMAKE_VARIABLE)
```

It's inconsistent, but I think the reason for the inconsistency is clear (same way you use single-quotes if you need a double-quote in the string, for instance)

Reply:
Good idea. Done.